src/Core/Validation/ValidationResult.php

```diff
@@ -50,204 +52,185 @@ class ValidationResult
     /**
      * Is the result valid or not.
      * Note that there can be non-error messages in the list.
-     *
-     * @var bool
      */
-    protected $isValid = true;
+    protected bool $isValid = true;
 
     /**
      * List of messages
-     *
-     * @var array
      */
-    protected $messages = [];
+    protected array $messages = [];
+
+    /**
+     * The class of the DataObject being validated.
+     */
+    private string $dataClass = '';
+
+    /**
+     * The record ID of the object being validated.
+     */
+    private int $recordID = 0;
```

Comment:
In various places we allow `mixed` for ID (e.g. see the recent updated to `SS_List` and related, which have `mixed $id` for the `byID()` method). We should do the same here - there's no guarantee that all projects use numeric IDs for all of their models.

Someone might have a model (not a subclass of `DataObject`) that integrates with a third-party service and uses a GUID - and they may want to use a `ValidationResult` with that model.

```suggestion
    private mixed $recordID = null;
```

Please also update the constructor, setter, and getter.

Reply:
Updated